File: FunFinance.API/Models/Budget.cs
using System.ComponentModel.DataAnnotations;

namespace FunFinance.API.Models
{
    public class Budget
    {
        public int Id { get; set; }

        [Required, StringLength(50)]
        public required string Name { get; set; }

        [Required]
        public decimal Limit { get; set; }

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        // Зовнішній ключ
        public int FamilyId { get; set; }

        // Навігаційні властивості
        public required virtual Family Family { get; set; }
        public virtual ICollection<Expense> Expenses { get; set; } = new List<Expense>();
    }
}

File: funfinance-frontend/src/models/budget.ts
export interface Budget {
    id: number;
    name: string;
    limit: number;
    startDate: string;
    endDate: string;
    familyId: number;
    expenses?: Expense[];
  }
  
  export interface BudgetCreate {
    name: string;
    limit: number;
    startDate: string;
    endDate: string;
    familyId: number;
  }
  
  export interface BudgetUpdate {
    name?: string;
    limit?: number;
    startDate?: string;
    endDate?: string;
  }
  
  // Імпортуємо Expense з моделі expense
  import { Expense } from './expense';